### PR TITLE
Updated ActivityResults to the new API recommended by Google

### PR DESCRIPTION
**Pull request by CristianMG**
Nothing special I'm participating in Hacktoberfest, a little contribution.

### CODE REVIEW

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 1]
```diff
@@ -94,6 +96,34 @@ public class MainActivity extends AegisActivity implements EntryListView.Listene
 
     private ActionMode.Callback _actionModeCallbacks = new ActionModeCallbacks();
 
+    ActivityResultLauncher<Intent> scanLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(),
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:99]: Now that we're using the new API, the ``CODE_`` request codes above are no longer used and can be removed.

**Review by alexbakker — CHANGES_REQUESTED** [review]
Thanks for working on this!

I found a few regressions while taking an initial look. For example, discarding changes and closing the EditEntryActivity when creating a new entry crashes the app. Same goes for existing entries. There may be more.

**Comment by CristianMG** [comment]
mmmmm I have a problem, one of the test is not passing.
testOverall()
`
java.lang.ArrayIndexOutOfBoundsException: length=10; index=-1
at java.util.ArrayList.set(ArrayList.java:455)
at com.beemdevelopment.aegis.ui.views.EntryAdapter.replaceEntry(EntryAdapter.java:193)
at com.beemdevelopment.aegis.ui.views.EntryListView.replaceEntry(EntryListView.java:379)
at com.beemdevelopment.aegis.ui.MainActivity.onEditEntryResult(MainActivity.java:321)
at com.beemdevelopment.aegis.ui.MainActivity.lambda$new$2$com-beemdevelopment-aegis-ui-MainActivity(MainActivity.java:106)
at com.beemdevelopment.aegis.ui.MainActivity$$ExternalSyntheticLambda7.onActivityResult(Unknown Source:4)
at androidx.activity.result.ActivityResultRegistry$1.onStateChanged(ActivityResultRegistry.java:149)
at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:360)
at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:271)
at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:313)
at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:151)
at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:134)
at androidx.lifecycle.ReportFragment.dispatch(ReportFragment.java:68)
at androidx.lifecycle.ReportFragment.dispatch(ReportFragment.java:144)
at androidx.lifecycle.ReportFragment.onStart(ReportFragment.java:109)
at android.app.Fragment.performStart(Fragment.java:2637)
at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1312)
at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1549)
at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1611)
at android.app.FragmentManagerImpl.dispatchMoveToState(FragmentManager.java:3045)
at android.app.FragmentManagerImpl.dispatchStart(FragmentManager.java:3002)
at android.app.FragmentController.dispatchStart(FragmentController.java:190)
at android.app.Activity.performStart(Activity.java:7206)
at android.app.Activity.performRestart(Activity.java:7332)
at android.app.Activity.performResume(Activity.java:7345)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3763)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3828)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1746)
at android.os.Handler.dispatchMessage(Handler.java:105)
at androidx.test.espresso.base.Interrogator.loopAndInterrogate(Interrogator.java:14)
at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:8)
at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:1)
at androidx.test.espresso.base.UiControllerImpl.injectMotionEvent(UiControllerImpl.java:5)
at androidx.test.espresso.action.MotionEvents.sendUp(MotionEvents.java:7)
at androidx.test.espresso.action.MotionEvents.sendUp(MotionEvents.java:1)
at androidx.test.espresso.action.Tap.sendSingleTap(Tap.java:5)
at androidx.test.espresso.action.Tap.-$$Nest$smsendSingleTap(Unknown Source:0)
at androidx.test.espresso.action.Tap$1.sendTap(Tap.java:3)
at androidx.test.espresso.action.GeneralClickAction.perform(GeneralClickAction.java:6)
at androidx.test.espresso.ViewInteraction$SingleExecutionViewAction.perform(ViewInteraction.java:2)
at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:25)
at androidx.test.espresso.ViewInteraction.-$$Nest$mdoPerform(Unknown Source:0)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:7)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.Handler.handleCallback(Handler.java:789)
at android.os.Handler.dispatchMessage(Handler.java:98)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6938)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
`
It's seem to be related with another problem. What do you think? @alexbakker

**Comment by alexbakker** [comment]
I believe https://github.com/beemdevelopment/Aegis/pull/998#discussion_r990626859 is a fix for that.

**Comment by Hacker437** [comment]
#807 This is the same thing I thought of doing in other places also

**Comment by alexbakker** [comment]
What's the status here?

**Comment by alexbakker** [comment]
Since there hasn't been any activity here for a while, I'm closing this one to clean the up the list of open pull requests a bit.

**Pull request closed without merging**